llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -458,6 +464,13 @@ Value *VPInstruction::generate(VPTransformState &State) {
   }
 
   switch (getOpcode()) {
+  case Instruction::PHI: {
```

Comment:
Somewhere it should be documented that VPInstructions of PHI opcode are used to model header phi's only.

Reply:
Added a comment here, thanks

Reply:
Thanks! Perhaps worth an assert? VPInstructions with phi opcodes could be used for non-header phi's prior to predication. Eventually also for uniform branches, and out-of-loop branches.